src/Type/Php/ArrayKeyExistsFunctionTypeSpecifyingExtension.php

```diff
@@ -57,13 +58,37 @@ public function specifyTypes(
 		$array = $node->getArgs()[1]->value;
 		$keyType = $scope->getType($key);
 		$arrayType = $scope->getType($array);
-		if (!$keyType instanceof ConstantIntegerType && !$keyType instanceof ConstantStringType && !$arrayType->isIterableAtLeastOnce()->no()) {
+
+		if (!$keyType instanceof ConstantIntegerType
+			&& !$keyType instanceof ConstantStringType
+			&& !$arrayType->isIterableAtLeastOnce()->no()) {
 			if ($context->truthy()) {
+				$arrayKeyType = $arrayType->getIterableKeyType();
+				if ((new StringType())->isSuperTypeOf($keyType)->yes()) {
+					$arrayKeyType = $arrayKeyType->toString();
```

Comment:
Jfyi we have `isString()` already too which I personally prefer. But it should come down to the same thing I guess

Reply:
I changed